thinkhazard_processing/decisiontree.py

```diff
@@ -0,0 +1,98 @@
+from thinkhazard_common.models import (
+    DBSession,
+    AdminLevelType,
+    )
+
+
+def apply_decision_tree(dry_run=False):
+    connection = DBSession.bind.connect()
+    trans = connection.begin()
+    try:
+        print "Purging previous relations"
+        connection.execute(clearall_query())
+        print "Calculating level REG"
+        connection.execute(level_REG_query())
+        print "Upscaling to PRO"
+        connection.execute(upscaling_query(u'PRO'))
+        print "Upscaling to COU"
+        connection.execute(upscaling_query(u'COU'))
+        if dry_run:
+            trans.rollback()
+        else:
+            trans.commit()
+    except:
+        trans.rollback()
+        raise
+
+
+def clearall_query():
+    return '''
+DELETE FROM datamart.rel_hazardcategory_administrativedivision;
+'''
+
+
+def level_REG_query():
+    return '''
+INSERT INTO datamart.rel_hazardcategory_administrativedivision (
+    administrativedivision_id,
+    hazardcategory_id,
+    source
+)
+SELECT DISTINCT
+    o.admin_id AS administrativedivision_id,
+    first_value(hc.id) OVER w AS hazardcategory_id,
+    first_value(hs.id) OVER w AS source
```

Comment:
From the thor mailing list: we should use the ```owner.organization``` field from the GeoNode API instead.

```json
{
    "owner": {
        "organization": "WorldBank",
        "...": "..."
```

This will be done in another PR, after this one is merged.